tests/unit/template_cassdc_test.go

```diff
@@ -423,11 +431,83 @@ var _ = Describe("Verify CassandraDatacenter template", func() {
 				KubectlOptions: defaultKubeCtlOptions,
 			}
 
-			err := renderTemplate(options)
+			renderedErr, _ := renderTemplate(options)
+
+			Expect(renderedErr).ToNot(BeNil())
+			Expect(renderedErr.Error()).To(ContainSubstring("set resource limits/requests when enabling allowMultipleNodesPerWorker"))
+
+		})
+
+		It("setting JVM heap settings at cluster-level only", func() {
+
+			dcName := "dc1"
+			options := &helm.Options{
+				SetValues: map[string]string{
+					"cassandra.datacenters[0].name": dcName,
+					"cassandra.datacenters[0].heap": "",
+					"cassandra.heap.size":           "100M",
+					"cassandra.heap.newGenSize":     "200M",
+				},
+				KubectlOptions: defaultKubeCtlOptions,
+			}
+
+			Expect(renderTemplate(options)).To(Succeed())
+
+			var config Config
+			Expect(json.Unmarshal(cassdc.Spec.Config, &config)).To(Succeed())
+			Expect(config.JvmOptions).ToNot(BeNil())
+			Expect(config.JvmOptions.InitialHeapSize).To(Equal("100M"))
+			Expect(config.JvmOptions.MaxHeapSize).To(Equal("100M"))
+			Expect(config.JvmOptions.YoungGenSize).To(Equal("200M"))
+		})
+
+		// Note: currently only one DC supported, to be expanded in future release.
+		It("setting JVM heap settings at dc level overriding cluster level", func() {
+
+			dcName := "dc1"
+			options := &helm.Options{
+				SetValues: map[string]string{
+					"cassandra.heap.size":                      "100M",
+					"cassandra.heap.newGenSize":                "200M",
+					"cassandra.datacenters[0].heap.size":       "300M",
+					"cassandra.datacenters[0].heap.newGenSize": "600M",
+					"cassandra.datacenters[0].name":            dcName,
+				},
+				KubectlOptions: defaultKubeCtlOptions,
+			}
+
+			Expect(renderTemplate(options)).To(Succeed())
+
+			var config Config
+			Expect(json.Unmarshal(cassdc.Spec.Config, &config)).To(Succeed())
+			Expect(config.JvmOptions).ToNot(BeNil())
+			Expect(config.JvmOptions.InitialHeapSize).To(Equal("300M"))
+			Expect(config.JvmOptions.MaxHeapSize).To(Equal("300M"))
+			Expect(config.JvmOptions.YoungGenSize).To(Equal("600M"))
+		})
+
+		// Note: currently only one DC supported, to be expanded in future release.
+		It("setting JVM heap settings at datacenter WITHOUT cluster-level settings", func() {
```

Comment:
If we define cluster-level defaults in `values.yaml`, this test case may not be relevant.

Reply:
👍